src/server/search/doc_index.cc

```diff
@@ -185,15 +185,22 @@ void ShardDocIndex::Rebuild(const OpArgs& op_args, PMR_NS::memory_resource* mr)
   auto cb = [this](string_view key, BaseAccessor* doc) { indices_.Add(key_index_.Add(key), doc); };
   TraverseAllMatching(*base_, op_args, cb);
 
+  was_built_ = true;
```

Comment:
what if we start dragonfly without loading a snapshot?
Do we still call `Rebuild` in this case?

Reply:
We can Rebuild upon FT.CREATE but only if it's in ACTIVE state:
https://github.com/dragonflydb/dragonfly/blob/main/src/server/search/doc_index.cc#L302